edb/server/protocol/protocol.pyx

```diff
@@ -706,6 +715,52 @@ cdef class HttpProtocol:
         response.status = http.HTTPStatus.BAD_REQUEST
         response.close_connection = True
 
+    async def _check_http_auth(
+        self,
+        HttpRequest request,
+        HttpResponse response,
+        str dbname,
+    ):
+        dbindex: dbview.DatabaseIndex = self.tenant._dbindex
+
+        try:
+            user = 'edgedb'  # I guess?
```

Comment:
I think this should be the `"sub"` in the JWT token or the username in the `Basic` authorization value, or `cfg::Auth.user` won't work if it's set to non-`edgedb` values. But that's probably an acceptable unimplemented feature at this moment, because this endpoint wasn't protected at all.

Reply:
`sub` isn't always present in the JWTs we generate, though. Ones for local instances have no `sub` but do have `"edgedb.server.any_role"` set.

Our cloud tokens do have a `sub` set, and it seems to be a random alphanumeric string; does it actually correspond to a role in edgedb?

Reply:
`sub` in the cloud token is the id of the cloud user, no relation to role, as those are specified in token scope instead.  We have the `X-EdgeDB-User` header to communicate the desired role instead.  Default to `edgedb` if it isn't present I guess.